pkg/apiserver/resthandler.go

```diff
@@ -147,7 +147,20 @@ func getRequestOptions(req *restful.Request, scope RequestScope, kind string, su
 		newQuery[subpathKey] = []string{req.PathParameter("path")}
 		query = newQuery
 	}
-	return queryToObject(query, scope, kind)
+	versioned, err := scope.Creater.New(scope.ServerAPIVersion, kind)
+	if err != nil {
+		// programmer error
+		return nil, err
+	}
+	if err := scope.Codec.DecodeParametersInto(query, versioned); err != nil {
```

Comment:
I think `[]byte` data passed into DecodeInto is required to have kind/apiVersion info embedded that allows the codec to know what conversions need to be done in order to get the data into the passed object. Query parameter data does not have that identifying info, which is why this is decoding into a versioned object (though I'd argue ServerVersion is the wrong version) then doing a conversion into an internal (or eventually unversioned?) object itself